docs/modules/serialization/pages/serialization.adoc

```diff
@@ -1,32 +1,75 @@
 = Serialization
+:description: To be able to send data over a network between cluster members and/or clients, you have to serialize it into raw bytes. Hazelcast has many serialization methods to choose from, depending on which Hazelcast topology you are using and what you plan on doing with your data.
 
-Hazelcast needs to serialize the Java objects that you put into Hazelcast because
-Hazelcast is a distributed system. The data and its replicas are
-stored in different partitions on multiple cluster members.
-The data you need may not be present on the local member, and in that case,
-Hazelcast retrieves that data from another member. This requires serialization.
+{description}
 
-Serialization is used in the following cases:
+== Serialization in Client/Server Mode
 
-* Adding key/value objects to a map
-* Putting items in a queue/set/list
-* Sending a runnable using an executor service
-* Processing an entry within a map
-* Locking an object
-* Sending a message to a topic
+In xref:overview:topology.adoc[client/server mode], you can serialize your data on the client side and/or the xref:ROOT:glossary.adoc#member[member] (server) side.
 
-Hazelcast optimizes the serialization for the basic types and their array types
-with its built-in serializers. By default, you cannot override this behavior; however,
-Hazelcast offers a way to override them using a configuration option,
-see the xref:overriding-built-in-serializers.adoc[Overriding the Default Serializers section].
-
-Hazelcast's built-in serializers can handle the following types:
+If you just want to store data on members, clients can handle serialization without the members needing to know how to do so. For example, you can serialize data as JSON and add it to a map. Then you can later request that information and deserialize on the client.
+
+However, in the following cases, you must configure members to be able to serialize and deserialize your data:
+
+[cols="a,a,a,a"]
+|===
+|Case|Clients|Reason|Exceptions
+
+|When the data structure's values are xref:data-structures:map.adoc#setting-in-memory-format[stored as objects]
+|All clients
+|Members must be able to serialize or deserialize the object in order to send it to other members or receive it from other members
+|
+
+|When you want to xref:computing:entry-processor.adoc[process map entries] or use an xref:computing:executor-service.adoc[executor service]
+|All clients
+|Members need to serialize the tasks
+|When using the Java client with xref:implementing-portable-serialization.adoc[`Portable`] serialization, you don't need to configure serialization on the member. For details, see xref:clusters:accessing-domain-objects.adoc[].
+
+|When you use xref:sql:querying-imap.adoc#key-and-value-fields[SQL queries on nested fields] whose values are not primitives
+|Java only

```

Comment:
Clients column is confusing for me. I think, it is not really clear what is meant there. 
We may rename it like `Supported Clients` OR
what if we completely remove it ?
Sql support is being added to other clients one by one. This document will be stale soon. 
Which clients support what can be looked up elsewhere. The purpose of this document is not that right?